Sources/KVKCalendar/CalendarModel.swift

```diff
@@ -542,9 +542,17 @@ public protocol CalendarDelegate: AnyObject {
     /// drag & drop events and resize
     func didChangeEvent(_ event: Event, start: Date?, end: Date?)
     
-    /// Controls whether event can be added
+    /** The method is **DEPRECATED**
+        Use a new **willAddNewEvent(_: _:)** that returns `Event?`
+     */
+    @available(*, deprecated, message: "Use the `willAddNewEvent(_: _:)` method that returns `Event?` instead.")
```

Comment:
This should be a relatively large change.

For users who have not used this method before, there is no impact.

But for users who have already used this method, they can only receive a warning, and then the previous logic will be invalid. I haven't thought of a better way to remind users to make changes.

Maybe it would be more appropriate to mark it as `@available(*, unavailable, message: "...")`?

Reply:
I suppose, we can use `obsoleted`here to show error with replace function

Reply:
> I suppose, we can use `obsoleted`here to show error with replace function

```swift
// ⬇️ ❌: Protocol members can only be marked unavailable in an @objc protocol
@available(swift, deprecated: 0.6.26, obsoleted: 0.6.27, message: "Use the `willAddNewEvent(_: _:)` method that returns `Event?` instead.")
func willAddNewEvent(_ event: Event, _ date: Date?) -> Bool
```

This approach may not be suitable for protocol

Reply:
hmm, let me think about it

Reply:
I implemented a compatible solution, the process is as follows:

![未标题-1 拷贝](https://github.com/user-attachments/assets/87bd8a7f-00c5-46c9-83fc-ed1952afa02d)

The old `willAddNewEvent` still works.

Reply:
Emm, I made a mistake. This change doesn’t seem to be any better for users. The update is still not painless, and the old `willAddNewEvent` still won’t take effect.